src/main/repositories/product_repository.py

```diff
@@ -0,0 +1,25 @@
+# This file has the product repository
+
+from src.main.repositories.generic_repository import GeneralRepository
+
+
+class ProductRepository(GeneralRepository):
+
+    def __init__(self):
+        self.products = {}
+
+    def add(self, product):
+        self.products[product.id] = product
+        return product
+
+    def get_by_id(self, product_id):
+        return self.products[product_id]
+
+    def get_all(self):
+        return self.products
```

Comment:
Do you want to return a map of products or a list of products here?

Reply:
fixed